### DISCUSSION TITLE
Is there any way to use Framework syntax in Framework Component slot used in .astro

### DISCUSSION THREAD

**Discussion by f3oall** [withastro/roadmap#318] [Proposal]
For example I have this component:
```vue
// NewsletterForm.vue
<template>
  <CustomForm  v-slot="{ fields }">
    <input v-model="fields.email" type="email" name="email" placeholder="your@email.com" aria-label="Email" />
    <button class="btn btn-info" type="submit">Submit</button>
  </CustomForm>
</template>

<script setup lang="ts">
import CustomForm from '@/components/Custom/Form/index.vue'
</script>
```
```astro
---
// page.astro
import NewsletterForm from './NewsletterForm.vue'
---
<div>
  <NewsletterForm client:visible/>
</div>
```

Currently, I am forced to make a separate file each time I need to use a component with Slot (that requires dynamic data).

I'm wondering whether there is a better way.

Ideally, I would like to have something like this:
```Astro
---
import CustomForm from '@/components/Custom/Form/index.vue'
---
<div>
  <FrameworkVue> <!-- Hint for compiler -->
     <CustomForm client:visible  v-slot="{ fields }">
        <input v-model="fields.email" type="email" name="email" placeholder="your@email.com" aria-label="Email" />
        <button class="btn btn-info" type="submit">Submit</button>
      </CustomForm>
  </FrameworkVue>
</div>
```

So I would call this "Inline ilands".

**Comment by jasikpark** [comment]
basically, "how can i hydrate this component and all of its children in the same component root"?

it's useful to have hydrated components that have unhydrated children, but it sounds like this would be very useful as well.

**Reply by f3oall** [reply]
Thanks for your answer @jasikpark. Yeah, I totally understand why it's useful to have unhydrated children. but sometimes you need the opposite (as in my example). Is there any chance that this topic can become an RFC?

**Comment by matthewp** [comment]
I don't think we can do this because this means we've have to know how to compile every frameworks syntax.

**Reply by jasikpark** [reply]
time for astro macros like rustlang proc macros 🤡
